generators/bootstrap-application-base/generator.ts

```diff
@@ -452,6 +452,15 @@ export default class BootstrapApplicationBase extends BaseApplicationGenerator {
         };
         entity.hasCyclicRequiredRelationship = detectCyclicRequiredRelationship(entity, new Set());
       },
+      processDerivedPrimaryKeyFields({ entity }) {
+        const primaryKey = entity.primaryKey;
+        if (!primaryKey || primaryKey.composite || !primaryKey.derived) {
+          return;
+        }
+        // derivedPrimary uses '@MapsId', which requires for each relationship id field to have corresponding field in the model
+        const derivedFields = primaryKey.derivedFields;
+        entity.fields.unshift(...derivedFields);
+      },
```

Comment:
It's implemented in server since we don't know if it's required in others servers.
`@MapsId` should be sql only.

Reply:
I guess that there could be an incidence of setting composite key on some `client` generators too.
The information of a key of type `primary`, `composite` or `derived` is not really a server concept but more an `application` one (i.e. at client level, one or more input will be greyed in the form).
The fact that a composite or derived key materializes to `MapsId` on server side is one the consequence setting this property at the entity-configuration level.
I'll remove the comment of MapsId if you agree with this principle.

Reply:
please reopen if need discussion, just to close all current comments

Reply:
I moved it to bootstrap application instead. TBH I do not understand well the difference between `base`, `bootstrap` and `nosuffix` as there are some surprising bits in base (i.e. changelog): may you please uncipher?

It's something common to client and server and its part of the entity's metamodel (not related to any technology, nor backend nor frontend), which is why I moved it in bootstrap-application which seems to make sense.

Also answering with 'no' without any suggestion, explanation or counterargument is not fair at all and is a waste of time for all of us. I won't rollback these bits until we agree on a relevant place for this logic (which won't be server because not related to backend tech in any mean).

Reply:
Duplicated the code and deprecated the 'server version'

Reply:
Done as asked